### DISCUSSION TITLE
self-hosted runner installed on Linux, doesn't pick job once ssh session terminates

### DISCUSSION THREAD

**Discussion by manjitsmakkar23** [community/community#57817] [New to GitHub] [answered]
I have a self-hosted runner installed on Linux VM and it's working fine until I have an active SSH session to it once the ssh session terminates, it doesn't pick up any job and Github shows waiting for the runner to pick the job and status remains in the queue. once I relogin into a runner, I can see the listener process is still running and if try to start the process again using ./run.sh it says already established connection. please advise how to fix the issue. ideally, the runner should be always listening to jobs even if no ssh session to it.

**Comment by Gustavonn07** [comment, marked as answer]
I searched for a while and found this hope it helps:

By default, when you disconnect from an SSH session, any processes started within that session may be terminated. To ensure that your self-hosted runner continues to run even after you disconnect from the SSH session, you can use a tool like nohup or tmux.

Here's a step-by-step guide on how to use nohup to keep the runner process running:

1. SSH into your Linux VM.
2. Navigate to the directory where your runner is installed.
3. Instead of running ./run.sh directly, use the nohup command to start the process:

`nohup ./run.sh &`

The nohup command allows the process to continue running even after the SSH session is terminated. The & at the end runs the process in the background.

4. You should see a message like "nohup: ignoring input and appending output to 'nohup.out'" indicating that the process has started successfully.
5. Now you can safely disconnect from the SSH session without affecting the runner process.
With this setup, the self-hosted runner should remain active and available to pick up jobs, even when you're not actively connected via SSH. Make sure to check the nohup.out file for any logs or error messages related to the runner process.

If you prefer to use tmux, it provides a more advanced terminal multiplexer that allows you to create and manage multiple terminal sessions within a single SSH session. It enables you to detach from and reattach to sessions, keeping your runner process running even when you're not connected.

**Reply by nanwp** [reply]
how to stop process nohup?

**Reply by ovyan** [reply]
@nanwp find it using ps aux | grep and kill the process
